### DISCUSSION TITLE
Mark annotation

### DISCUSSION THREAD

**Discussion by dorinemerlat** [jokergoo/ComplexHeatmap#720] [Q&A]
Hi, I have a little issue with the mark annotations. I can add them to my heatmap but I would like to use two different colors following the fact that the genes are known or not.  To add them I used that : 
```
annot = rowAnnotation(foo = anno_mark(
    at = c(1, 4, 8),
    labels =  c("ALDH1A1", "UPTR", "CCNO"),
    labels_gp = gpar(col = "black", fontsize = 10)))
```

To obtain two different colors for annotation I thought I would use the list() in the different fields (like for the other annotation types), but it didn't work (error : `at` should be numeric row index corresponding to the matrix) : 
```
annot = rowAnnotation(foo = anno_mark(
    at = list(new = c(1), known = c(4,8)),
    labels = list(new = c("ALDH1A1"), known = c("UPTR", "CCNO")),
    labels_gp = list(new = gpar(col = "black", fontsize = 10), known = gpar(col = "black", fontsize = 10))))
```

Is there any other way to put the annotations in different colors ?

Thanks :)

**Comment by jokergoo** [comment]
You can set `col` as a vector `in `labels_gp`:

```r
annot = rowAnnotation(foo = anno_mark(
    at = c(1, 4, 8),
    labels =  c("ALDH1A1", "UPTR", "CCNO"),
    labels_gp = gpar(col = c("black", "red", "red"), fontsize = 10)))

Heatmap(matrix(rnorm(100), 10)) + annot
```

![image](https://user-images.githubusercontent.com/449218/114294649-549dba80-9aa0-11eb-944c-42a52410a227.png)

**Reply by dorinemerlat** [reply]
thank you, it works perfectly !
